frontend/src/lib/WebsocketConnection.tsx

```diff
@@ -617,18 +629,22 @@ export function doHealthPing(
   }
 
   connect = () => {
-    const uri = uriList[uriNumber]
-    logMessage(LOG, `Attempting to connect to ${uri}.`)
+    const uriParts = uriPartsList[uriNumber]
+    const healthzUri = buildHttpUri(uriParts, SERVER_PING_PATH)
+    const allowedOriginsUri = buildHttpUri(uriParts, ALLOWED_ORIGINS_PATH)
+
+    logMessage(LOG, `Attempting to connect to ${healthzUri}.`)
 
     if (uriNumber === 0) {
       totalTries++
     }
 
-    axios
-      .get(uri, {
-        timeout: minimumTimeoutMs,
-      })
-      .then(() => {
+    Promise.all([
+      axios.get(healthzUri, { timeout: minimumTimeoutMs }),
+      axios.get(allowedOriginsUri, { timeout: minimumTimeoutMs }),
+    ])
```

Comment:
I think this makes sense to me but it's not obvious why these two fetches are run in parallel. (My assumption is "because it doesn't really change anything and is faster than running them sequentially" - but I don't quite understand what the `healthz` fetch is getting us here?) Do you mind adding a small comment explaining why this connect func needs to call both `healthz` _and_ `allowOrigins`?

Reply:
Ack -- will add an explanatory comment here